### DISCUSSION TITLE
Block IP based on Username and IP

### DISCUSSION THREAD

**Discussion by leahoswald** [fail2ban/fail2ban#3850] [Q&A] [answered]
Hi,

is it possible with fail2ban to have buckets not only for IPs but for IPs and usernames gathered for example with a F-Variable. What I want to do is block an IP if, for example, it is failing to login for 10 times with a specific username but not if it does 10 login attempts for 10 different users. [This is an simplified example, see answer below]

**Comment by sebres** [comment, marked as answer]
Yes, it is possible.
You can capture the user name using `<F-USER>...</F-USER>` in RE... And then use `<F-USER>` with `ignorecommand` to check the ticket must be ignored or not.

> but not if it does 10 login attempts for 10 different users.

Strange and questionable approach... 10 login attempts for 10 different users is basically a standard attack vector.
Normally often requested vice versa approach - ignore (legitimate) users trying single user (with almost the same password), e. g. if user forgot/changed his password and some program still trying with old password - this is possible with logging password hash and capturing it with `<F-PWDHASH>...</F-PWDHASH>` and comparing them with previous value in `ignorecommand`.

**Reply by leahoswald** [reply]
>  Strange and questionable approach...

The actual use case is another one, but I thought this is easier to explain. Sorry for that. The actual use case is a combined logfile for multiple domains where I want to filter some bots but on a per-domain basis without creating a jail for every domain. I would like to allow some bots a few requests per domain, what could be easier than finding a good global value.

Use case is the facebookexternalhit bot, that is used for scraping and for link preview. So I need a way to allow a few request but block the high volume scraping.

**Reply by leahoswald** [reply]
Oh and thanks! Your solution should work.
